app/org/sagebionetworks/bridge/play/controllers/ConsentController.java

```diff
@@ -25,36 +26,41 @@
     private ParticipantOptionsService optionsService;
 
     @Autowired
-    public void setConsentService(ConsentService consentService) {
+    final void setConsentService(ConsentService consentService) {
         this.consentService = consentService;
     }
     @Autowired
-    public void setOptionsService(ParticipantOptionsService optionsService) {
+    final void setOptionsService(ParticipantOptionsService optionsService) {
         this.optionsService = optionsService;
     }
 
+    @Deprecated
     public Result getConsentSignature() throws Exception {
         final UserSession session = getAuthenticatedAndConsentedSession();
         final Study study = studyService.getStudy(session.getStudyIdentifier());
-
-        ConsentSignature sig = consentService.getConsentSignature(study, session.getUser());
-        return ok(ConsentSignature.SIGNATURE_WRITER.writeValueAsString(sig));
+        return getConsentSignatureV2(study.getIdentifier());
     }
 
     @Deprecated
     public Result giveV1() throws Exception {
-        return giveConsentForVersion(1);
+        final UserSession session = getAuthenticatedSession();
+        final Study study = studyService.getStudy(session.getStudyIdentifier());
+        return giveConsentForVersion(1, SubpopulationGuid.create(study.getIdentifier()));
     }
 
+    @Deprecated
     public Result giveV2() throws Exception {
-        return giveConsentForVersion(2);
+        final UserSession session = getAuthenticatedSession();
+        final Study study = studyService.getStudy(session.getStudyIdentifier());
+        return giveV3(study.getIdentifier());
```

Comment:
Why call give3() which calls giveForVersion(2) when you can call giveForVersion(2) directly?